src/ast/passes/semantic_analyser.cpp

```diff
@@ -216,6 +213,300 @@ class SemanticAnalyser : public Visitor<SemanticAnalyser> {
 
 } // namespace
 
+// clang-format off
+enum class AssignWant {
+  MAP,
+  VAR,
+  NONE,
+};
+
+struct arg_type_spec {
+  Type type = Type::integer;
+  bool literal = false;
+  // This indicates that this is just a placeholder
+  // as we use the index in the vector of arg_type_spec
+  // as the number argument to check
+  bool skip_check = false;
+};
+
+struct arg_spec {
+  size_t min = 0;
+  size_t max = 0;
+  bool check_assign = false;
+  AssignWant assign = AssignWant::NONE;
```

Comment:
Naming super-duper-nit (I just cannot help myself :grinning:):

Since the struct is called `arg_spec`, i.e. a specification of a builtin, having the word "check" there is kind of confusing. How about something like:
```suggestion
  bool has_return = false;
  ReturnKind return_kind = ReturnKing::NONE;
```

Reply:
Ah, well I'm also going to change `arg_spec` to `call_spec`. And yeah, good call, I'll make those fields refer to the return.